indicator.py

```diff
@@ -116,44 +128,95 @@ def _click_exit(self):
+            self.config_action.setDisabled(False)
+            pass
+
+        if message == 'ON':
+            self.indicator.setIcon(QIcon(self._get_file('./icons/on.png')))
+            self.indicator.setToolTip('ON')
+            self.disconnect_action.setDisabled(False)
+            pass
+
+        if message == 'OFF':
+            self.indicator.setIcon(QIcon(self._get_file('./icons/icon.png')))
+            self.indicator.setToolTip('OFF')
+            self.disconnect_action.setDisabled(True)
+            self.connect_action.setDisabled(False)
+            self.config_action.setDisabled(False)
+            pass
+
+    def _show_update_notification(self, flag):
+        if flag and self.indicator.supportsMessages:
+            self.indicator.showMessage(self.APP_NAME, 'There is a new update available', QIcon(self._get_file('./icons/icon.png')))
+
+
+class VPNThread(QThread):
+    status = Signal(str)
+    log = Signal(str)
+
+    def __init__(self, vpn_logs_file=None):
+        super().__init__(None)
+
+        self.vpn_logs_file = vpn_logs_file
+
+    def run(self):
         with open(self.vpn_logs_file.name) as f:
             while True:
                 line = f.readline()
+
+                if len(line) > 0:
+                    self.log.emit(line.rstrip())
+
                 if line.find('Error') != -1 or line.find('ERROR') != -1:
-                    self.indicator.setIcon(QIcon(self._get_file('./icons/err.png')))
-                    self.indicator.setToolTip('ERROR')
-                    self.connect_action.setDisabled(False)
-                    self.config_action.setDisabled(False)
+                    self.status.emit('ERR')
                     break
 
                 if line.find('Tunnel is up and running') != -1:
-                    self.indicator.setIcon(QIcon(self._get_file('./icons/on.png')))
-                    self.indicator.setToolTip('ON')
-                    self.disconnect_action.setDisabled(False)
-
+                    self.status.emit('ON')
 
                 if line.find('Logged out') != -1:
-                    self.indicator.setIcon(QIcon(self._get_file('./icons/icon.png')))
-                    self.indicator.setToolTip('OFF')
-                    self.disconnect_action.setDisabled(True)
-                    self.connect_action.setDisabled(False)
-                    self.config_action.setDisabled(False)
+                    self.status.emit('OFF')
                     break
 
-                sleep(0.1)
+                self.sleep(0.1)
 
-    def _get_file(self, file_path):
+
+class AppUpdateThread(QThread):
+    update_available = Signal(bool)
+
+    def run(self):
         try:
-            base = sys._MEIPASS
-        except Exception:
-            base = path.abspath('.')
+            response = urlopen('https://api.github.com/repos/tshiamobhuda/fortivpn-quick-tray-qt/releases/latest')
+            release = json.loads(response.read().decode())
+        except HTTPError as e:
```

Comment:
`as e` can be removed since `e` is not used
```suggestion
        except HTTPError:
```